bftengine/src/bftengine/ControllerWithSimpleHistory.cpp

```diff
@@ -93,18 +89,14 @@ bool ControllerWithSimpleHistory::onNewSeqNumberExecution(SeqNum n) {
   SeqNoInfo& s = recentActivity.get(n);
 
   Time now = getMonotonicTime();
-  auto duration = now - s.prePrepareTime;
+  auto duration = std::chrono::duration_cast<std::chrono::microseconds>(now - s.prePrepareTime);
 
   const auto MAX_DURATION_MICRO = microseconds(2 * 1000 * 1000);
   const auto MIN_DURATION_MICRO = microseconds(100);
 
-  if (duration > MAX_DURATION_MICRO)
-    s.durationMicro_ = MAX_DURATION_MICRO;
-  else if (duration < MIN_DURATION_MICRO)
-    s.durationMicro_ = MIN_DURATION_MICRO;
+  s.durationMicro_ = normalizeToRange(MIN_DURATION_MICRO, MAX_DURATION_MICRO, duration);
```

Comment:
`s.durationMicro_ = normalizeToRange(MIN_DURATION_MICRO, MAX_DURATION_MICRO, duration);` As you can see prior this change, if a duration was within the desired range, MIN_DURATION_MICRO and MAX_DURATION_MICRO, it was ignored, probably a line was omitted by mistake.
The byproduct of this bug is that not only the value is ignored, the duration set to 0, pushing the average to the minimum.

Reply:
Can you point out in the description, which test case reproduces this bug?